cs/ccxt/base/Exchange.Types.cs

```diff
@@ -1391,4 +1391,92 @@ public Account(object accountStructure2)
         type = Exchange.SafeString(accountStructure, "type");
         code = Exchange.SafeString(accountStructure, "code");
     }
+}
+
+
+public struct Option
+{
+    public string? currency;
+    public string? symbol;
+    public Int64? timestamp;
+    public string? datetime;
+
+    public double? impliedVolatility;
+
+    public double? openInterest;
+
+    public double? bidPrice;
+    public double? askPrice;
+    public double? midPrice;
+    public double? markPrice;
+    public double? lastPrice;
+    public double? underlyingPrice;
+    public double? change;
+    public double? percentage;
+
+    public double? baseVolume;
+    public double? quoteVolume;
+    public Dictionary<string, object> info;
+
+    public Option(object option2)
+    {
+        var ticker = (Dictionary<string, object>)option2;
+        currency = Exchange.SafeString(option2, "currency");
+        symbol = Exchange.SafeString(option2, "symbol");
+        timestamp = Exchange.SafeInteger(option2, "timestamp");
+        datetime = Exchange.SafeString(option2, "datetime");
+        impliedVolatility = Exchange.SafeFloat(option2, "bidVolume");
+        openInterest = Exchange.SafeFloat(option2, "openInterest");
+        bidPrice = Exchange.SafeFloat(option2, "askVolume");
+        askPrice = Exchange.SafeFloat(option2, "vwap");
+        midPrice = Exchange.SafeFloat(option2, "open");
+        markPrice = Exchange.SafeFloat(option2, "close");
+        lastPrice = Exchange.SafeFloat(option2, "last");
+        underlyingPrice = Exchange.SafeFloat(option2, "previousClose");
+        change = Exchange.SafeFloat(option2, "change");
+        percentage = Exchange.SafeFloat(option2, "percentage");
+        baseVolume = Exchange.SafeFloat(option2, "baseVolume");
+        quoteVolume = Exchange.SafeFloat(option2, "quoteVolume");
+        info = ticker.ContainsKey("info") ? (Dictionary<string, object>)ticker["info"] : null;
+    }
+}
+
+
+public struct OptionChain
+{
+    public Dictionary<string, object> info;
+    public Dictionary<string, Option> chains;
+
+    public OptionChain(object optionchains2)
+    {
+        var marginModes = (Dictionary<string, object>)optionchains2;
```

Comment:
is that supposed to be marginModes?

Reply:
I don't think so, I changed the variable names here: #23158